plugins/azure_sentinel/plugin.spec.yaml

```diff
@@ -0,0 +1,879 @@
+        title: Resource Group Name
+        required: true
+        description: The name of the resource group within the user's subscription. The name is case insensitive.
+        type: string
+      subscriptionId:
+        title: Subscription ID
+        required: true
+        description: "Azure subscription ID Regex pattern: ^[0-9A-Fa-f]{8}-([0-9A-Fa-f]{4}-){3}[0-9A-Fa-f]{12}$"
+        type: string
+      workspaceName:
+        title: Workspace Name
+        required: true
+        description: The name of the workspace
+        type: string
+      api-version:
+        title: API version
+        description: The API version to use for this operation.
+        required: true
+        type: string
+        default: 2021-04-01
+      filter:
+        title: filter the results
+        required: false
+        description: Filters the results, based on a Boolean condition. Optional.
+        type: string
+      orderBy:
+        title: order the results
+        required: false
+        description: sorts the results 
+        type: string
+      top:
+        title: Return top results
+        required: false
+        description: Return only n first results. 
+        type: integer
+    output:
+      bookmarks:
+        title: Bookmark List
+        description: All the bookmarks assigned to the given incident.
+        type: "[]HuntingBookmark"
+  list_entities:
+    title: List entities
+    description: Get all incidents entities
+    input:
+      incidentId:
+        title: Incident ID
+        required: true
+        description: Incident ID
+        type: string
+      resourceGroupName:
+        title: Resource Group Name
+        required: true
+        description: The name of the resource group within the user's subscription. The name is case insensitive.
+        type: string
+      subscriptionId:
+        title: Subscription ID
+        required: true
+        description: "Azure subscription ID Regex pattern: ^[0-9A-Fa-f]{8}-([0-9A-Fa-f]{4}-){3}[0-9A-Fa-f]{12}$"
+        type: string
+      workspaceName:
+        title: Workspace Name
+        required: true
+        description: The name of the workspace
+        type: string
+      api-version:
+        title: API version
+        description: The API version to use for this operation.
+        required: true
+        type: string
+        default: 2021-04-01 
+      filter:
+        title: filter the results
+        required: false
+        description: Filters the results, based on a Boolean condition. Optional.
+        type: string
+      orderBy:
+        title: order the results
+        required: false
+        description: sorts the results 
+        type: string
+      top:
+        title: Return top results
+        required: false
+        description: Return only n first results. 
+        type: integer
+    output:
+      entities:
+        title:  List
+        description: All the entities assigned to the given incident.
+        type: "[]object"
```

Comment:
Generic objects are too difficult to work with - can you type this specifically?

Reply:
I improved it a bit.